### ISSUE TITLE
Fix false positive for ::-moz-focus-inner in no-descending-specificity

### ISSUE THREAD

**Issue by cpmsmith** [stylelint/stylelint#5032]
> Clearly describe the bug

`::-moz-focus-inner` is treated as a pseudo-class, not a pseudo-element, and so can cause false positives in `no-descending-specificity`.

> Which rule, if any, is the bug related to?

`no-descending-specificity`

> What code is needed to reproduce the bug?

```css
a::-moz-focus-inner {}
a {}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "no-descending-specificity": true
  }
}
```

> Which version of stylelint are you using?

`v13.7.2`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Originally encountered running it with [SublimeLinter](https://github.com/SublimeLinter/SublimeLinter-stylelint), but reproducible in the [Stylelint demo page](https://stylelint.io/demo#N4Igxg9gJgpiBcICG94FoC2EBeaBmEYArgM5oCWAdpTAE4AEwAvgDqVKNMgA0IJAnpQAuSAB4JwJEj3ARKecgHMJwNvXosQtIgBsYJTfEZr1GkJQhpYJMDEpQqitCQAOMMOQUeh-Q-SHaMCaslFxMQA)

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

No errors to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following error was flagged:

```
2:1 error Expected selector "a" to come before selector "a::-moz-focus-inner" (no-descending-specificity)
```

### Fixing it

This could simply be fixed by adding `-moz-focus-inner` to the list of known pseudo-elements in [keywordSets.js](https://github.com/stylelint/stylelint/blob/master/lib/reference/keywordSets.js), but relying on that list is obviously not the greatest—`-moz-focus-inner` was actually the original example given for this bug way back in #1811, and the fix still missed it.

If we instead made the selector parsing assume any selector with a double colon is a pseudo-element, not a pseudo-class, that could be a more general fix, and would match [the spec](https://www.w3.org/TR/selectors-3/#pseudo-elements):

> This `::` notation is introduced by the current document in order to establish a discrimination between pseudo-classes and pseudo-elements.

Of course, the confusion between e.g. `:after` and `::after` is [well-known](https://css-tricks.com/to-double-colon-or-not-do-double-colon/), and the former is in (limited, but spec-compliant) use, so we would still need a keyword list, but this would solve all syntactically valid cases going forward, and existing logic would probably handle any common invalid ones gracefully.

**Comment by jeddy3** [comment]
@cpmsmith Thanks for the report, creating a demo and for using the template.

I agree with your more comprehensive suggested fix.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time. 

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.
